### DISCUSSION TITLE
sshfs

### DISCUSSION THREAD

**Discussion by WGroleau** [Homebrew/discussions#2491] [Everyday usage]
"Error: sshfs has been disabled because it requires closed-source macFUSE!"

Apparently not so.  I downloaded installers for MacFUSE and SSHFS from https://osxfuse.github.io.  MacFUSE made me accept a BSD license and SSHFS had a GPL License.  I read both far enough (no scrolling) to see that they are **open source**.

MacFUSE requires sudo privilege to install.  I think SSHFS did not but I don’t remember for sure.

SSHFS requires sudo to install in a system-owned mount-point but then the casual user cannot get into the directory.

sshfs  user@server:/…/path  /…/DIR worked (without sudo) when I was the owner of  DIR and gave me full access to everything in the shell.  Owner and group were mine locally and were still correct (different) on the server.  I edited and saved a text file, and it did not change owner or group on the server.

The MACOS Finder GUI also gives normal access to everything,  When mounted, the mount point …/UG in the GUI has a name change to “macFUSE volume 0 (sshfs)” but that is harmless, and it changes back when dismounted.   It still appears as DIR to “ls” in the shell.

**Comment by gromgit** [comment]
> I downloaded installers for MacFUSE and SSHFS from https://osxfuse.github.io. MacFUSE made me accept a BSD license and SSHFS had a GPL License. I read both far enough (no scrolling) to see that they are **open source**.

The macFUSE maintainer and many other macFUSE users disagree with your interpretation. Just three of the related issues filed on this matter:
* https://github.com/osxfuse/osxfuse/issues/768
* https://github.com/osxfuse/osxfuse/issues/771
* https://github.com/osxfuse/osxfuse/issues/841

If he changes his mind and actually releases the source, I'm sure the Homebrew maintainers would be more than happy to welcome formulae that depend on macFUSE back into the core. Until then, I'm maintaining these formulae over on my personal tap: https://github.com/gromgit/homebrew-fuse

**Comment by WGroleau** [comment]
Well, I'm sure the license said what I claimed.  I'm too lazy to download another installer to look again.  Anyway, since I'm only using it and not distributing, it doesn't matter to me.  Plus, it worked great after install, but is broken after reboot.

**Comment by carlocab** [comment]
> I'm sure the license said what I claimed.

No, you either misread or are misremembering. [Here](https://github.com/osxfuse/osxfuse/blob/a742dc201b6268f19a2c41b1c6860cffac605993/LICENSE.txt#L3-L20) is an extract of macFUSE's license:
```
  Copyright (c) 2011-2021 Benjamin Fleischer
  All rights reserved.

  Redistribution and use in source and binary forms, with or without
  modification, are permitted provided that the following conditions are met:

  1. Redistributions of source code must retain the above copyright notice, this
     list of conditions and the following disclaimer.
  2. Redistributions in binary form must reproduce the above copyright notice,
     this list of conditions and the following disclaimer in the documentation
     and/or other materials provided with the distribution.
  3. Neither the name of the copyright holder nor the names of its contributors
     may be used to endorse or promote products derived from this software 
     without specific prior written permission.
  4. Redistributions in binary form, bundled with commercial software, are not
     allowed without specific prior written permission. This includes the
     automated download or installation or both of the binary form in the
     context of commercial software.
```

It _looks_ like a BSD license, but it's **not** a BSD license. In particular, point 4 very explicitly states that macFUSE is not free or open source. Just writing down the license to make it look like a BSD license doesn't make it a BSD license. We also don't really need to know anything about licensing to to work out that macFUSE is not open-source. The source code isn't even available at the [GitHub repository](https://github.com/osxfuse/osxfuse).

**Reply by WGroleau** [reply]
The license the installer showed me had a centered title near the top.  #4 says nothing about source.  #1 does not forbid distribution of source, merely demands preserving the copyright.  Anyway, what matters to me is that it worked when I installed it.

**Reply by carlocab** [reply]
> #4 says nothing about source. #1 does not forbid distribution of source, merely demands preserving the copyright.

It seems we just have different understandings of the term "closed source". "Open source" is widely understood as meaning much more than source availability or the right to redistribute the source code. Perhaps a better term for "open source software" is "free software" (which is actually the language that the FSF uses), because that's what most people who understand the subject mean when they say "open source".

macFUSE is not free software because point 4 restricts what you can do with it.

In any case, if you're not happy with the wording of the message shown to you about macFUSE being closed-source, you should instead interpret it to say
```
Error: sshfs has been disabled because it requires macFUSE which has licensing restrictions on its use!
```

**Reply by gromgit** [reply]
It's actually even simpler than that: The macFUSE source code (at least at v4, and probably extending back to v3.9) is not publicly available, period. I don't know of any common interpretation of "open source" that includes "no source code for anyone!"

**Reply by WGroleau** [reply]
Well, again, it tells you how you must distribute source, and doesn't say you may not do so.  I'm not interested in doing so anyway.
